core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/project/ProjectResource.scala

```diff
@@ -348,6 +348,27 @@ class ProjectResource {
     }
   }
 
+  /**
+    * This method updates the description of a specified, existing project
+    *
+    * @param pid project ID
+    */
+  @POST
+  @Path("/{pid}/update/description")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def updateProjectDescription(@PathParam("pid") pid: UInteger, project: UserProject): Unit = {
+    val description = project.getDescription
+    val userProject = userProjectDao.fetchOneByPid(pid)
+    try {
```

Comment:
line 361 and line 362 you are fetching for the project from db, can there be exceptions?

Reply:
Did some more digging into this and found that the case I discussed with you where the pid is not in the table actually doesn't throw an exception, it just has a null value returned.